### PR TITLE
Remove more redundant tests: md4, md5, rmd, rc4, p5_crpt2

### PR DESCRIPTION
**Pull request by snhenson**
##### Description of change

As with #2783 thsi removes some stand alone tests where evptest.txt uses exactly the same data and uses the API in a similar way.

### CODE REVIEW

File: test/build.info [hunk 1]
```diff
@@ -70,22 +70,10 @@ IF[{- !$disabled{tests} -}]
...
-  DEPEND[md5test]=../libcrypto
-
   SOURCE[hmactest]=hmactest.c
   INCLUDE[hmactest]=../include
   DEPEND[hmactest]=../libcrypto
 
-  SOURCE[wp_test]=wp_test.c
-  INCLUDE[wp_test]=../include
-  DEPEND[wp_test]=../libcrypto
-
```
Comment (levitte) [inline test/build.info]: Huh?  This didn't go in the other PR?  Bummer!

Comment (snhenson) [inline test/build.info]: Nope, missed it.

Comment (levitte) [inline test/build.info]: Ah well, good thing you caught it now :-)

**Comment by dot-asm** [comment]
But rc4test.c does more than evptests.txt. And it did help to catch bugs that otherwise wouldn't be caught...

**Comment by snhenson** [comment]
Ah missed that part. I'll add a PR with a revert for rc4test.

**Comment by snhenson** [comment]
Done in #2789

**Comment by TimJSwan89** [comment]
So am I correct in stating that the functionality which was covered by `test/md4test.c` is covered by a subset of {`test/evp_extra_test.c`,`test/evp_test.c`,`test/evp_test.h`}?

**Comment by mattcaswell** [comment]
> So am I correct in stating that the functionality which was covered by test/md4test.c is covered by a subset of {test/evp_extra_test.c,test/evp_test.c,test/evp_test.h}?

Yes. Note this only applies to version 1.1.1/master. `md4test.c` still exists in 1.1.0 and below.

If you take a look at the md4test.c as it appears in 1.1.0:

https://github.com/openssl/openssl/blob/OpenSSL_1_1_0-stable/test/md4test.c

You can see that it basically runs md4 on 7 different inputs ("", "a", "abc", etc), and compares the calculated output with the expected output.

This same job is done by evp_test.c, but more generically, i.e. we use the same test for lots of different digests, ciphers, etc. The test vectors are read in from a file. You can see the md4 ones here:

https://github.com/openssl/openssl/blob/master/test/recipes/30-test_evp_data/evpdigest.txt#L260

In that file you can see that it has the exact same 7 test vectors as appeared in the original md4test.c ("", "a", "abc" etc)

**Pull request closed without merging**